config/config.go

```diff
@@ -792,6 +792,7 @@ type KubernetesSDConfig struct {
 	RetryInterval   model.Duration `yaml:"retry_interval,omitempty"`
 	RequestTimeout  model.Duration `yaml:"request_timeout,omitempty"`
 	TLSConfig       TLSConfig      `yaml:"tls_config,omitempty"`
+	Namespace 	string 	       `yaml:"namespace,omitempty"`
```

Comment:
@brian-brazil Yes, this namespace corresponds to the k8 namespace. To provide more context into this specific issue - all our applications and prometheus server are deployed in a particular namespace in Kubernetes. Since it is deployed under a namespace different from default, it has access only to the pods/services in the that namespace. Since discovery module is trying to access resources without namespace (e.g., "/api/v1/services"), this access fails.